numba/core/dispatcher.py

```diff
@@ -862,6 +864,19 @@ def _numba_type_(self):
     def enable_caching(self):
         self._cache = FunctionCache(self.py_func)
 
+    @functools.lru_cache()
+    def cache_deps_info(self, sig) -> pt.Dict[str, FileStamp]:
+        """Hash the code of its function, the closure variables and add them
+        to the respective hashes of all its function dependencies
+        """
+        sig_args = sig.args
+        if isinstance(self.overloads[sig_args].type_annotation, str):
+            # we assume this is because the function was loaded from cache
```

Comment:
```suggestion
            # We assume this is because the function was loaded from cache
            # When a function is cached, the type_annotation property is serialized
            # as a string. Ref:
            # https://github.com/numba/numba/blob/be990a818c377408a1f41c37c7720bc2ba0fb89a/numba/core/compiler.py#L207  # noqa: E501
```